### PR TITLE
Wrote Enzyme tests for "BackButton" component №418

### PR DESCRIPTION
**Pull request by 100vbaSasha**
Wrote Enzyme tests for BackButton component on admin panel

image

- [✔ ] 🔽 My branch is up-to-date with "development" branch
- [ ✔] ✅ All tests passed locally
- [✔ ] ✨ My changes working with up-to-date front-end and back-end part locally, like charm
- [✔ ] 🔗 Link pull request to issue

### CODE REVIEW

File: src/components/buttons/back-button/tests/back-button.test.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    component = mount(<BackButton {...testProps} />);
+  });
+  afterEach(() => {
+    component.unmount();
+  });
+  it('Should handle click the button', () => {
+    useDispatchMock.mockReturnValue(mockHolder);
+    const wrapper = mount(<BackButton />);
+    wrapper.find('button').simulate('click');
+    expect(mockHolder.mock.calls.length).toEqual(0);
```
Comment (fulljsdev) [inline src/components/buttons/back-button/tests/back-button.test.js:38]: mockHolder.mock.calls.length should be more than 0

**Review by fulljsdev — COMMENTED** [review]
mockHolder.mock.calls.length should be more than 0

File: src/components/buttons/back-button/tests/back-button.test.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  });
+  it('Should openSuccessSnackbar exist', () => {
+    const testProps = {
+      initial: true,
+      color: 'primary',
+      type: 'button',
+      variant: 'outlined'
+    };
```
Comment (AndriyLymych) [inline src/components/buttons/back-button/tests/back-button.test.js:46]: please, put test data in file variables

File: src/components/buttons/back-button/tests/back-button.test.js (hunk 1 above)
Comment (100vbaSasha) [reply src/components/buttons/back-button/tests/back-button.test.js]: changed

File: src/components/buttons/back-button/tests/back-button.test.js (hunk 2 above)
Comment (100vbaSasha) [reply src/components/buttons/back-button/tests/back-button.test.js]: Done

File: src/components/buttons/back-button/tests/back-button.test.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    component = mount(<BackButton {...testProps} />);
+  });
+  afterEach(() => {
+    component.unmount();
+  });
+  it('Should handle click the button', () => {
+    useDispatchMock.mockReturnValue(mockHolder);
+    const wrapper = mount(<BackButton />);
+    wrapper.find('button').simulate('click');
+    expect(mockHolder.mock.calls.length).toBeDefined();
```
Comment (fulljsdev) [inline src/components/buttons/back-button/tests/back-button.test.js:38]: wrapper.find('button').simulate('click') does not change mockHolder.mock.calls.length

File: src/components/buttons/back-button/tests/back-button.test.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+  afterEach(() => {
+    component.unmount();
+  });
+  it('Should handle click the button', () => {
+    useDispatchMock.mockReturnValue(mockHolder);
+    const wrapper = mount(<BackButton onClick={mockHolder} />);
+    wrapper.find('button').simulate('click');
+    expect(mockHolder.mock.calls.length).toEqual(1);
+  });
+  it('Should openSuccessSnackbar exist', () => {
```
Comment (fulljsdev) [inline src/components/buttons/back-button/tests/back-button.test.js:40]: This test case does not work properly

**Pull request closed without merging**